app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Service.java

```diff
@@ -50,6 +50,18 @@
     public static final UUID UUID_CHARACTERISTIC_AUTH = UUID.fromString("00000009-0000-3512-2118-0009af100700");
     public static final UUID UUID_CHARACTERISTIC_10_BUTTON = UUID.fromString("00000010-0000-3512-2118-0009af100700");
 
+    // Put UUIDs according to http://jellygom.com/2016/09/30/Mi-Band-UUID.html
+    // Also good resource is http://muhlenxi.com/2017/05/08/iOS-Bluetooth-Low-Energy-Develop-Chapter7/ however hard readable
+    // even using Google Translate translation as I don't know Chinese
+    // Update: Translation using following URL seems to be understandable:
+    // https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=http%3A%2F%2Fmuhlenxi.com%2F2017%2F05%2F08%2FiOS-Bluetooth-Low-Energy-Develop-Chapter7%2F
+    // Another good resource may be: http://allmydroids.blogspot.cz/2014/12/xiaomi-mi-band-ble-protocol-reverse.html
+    public static final UUID UUID_CHARACTERISTIC_DEV_INFO_SERIAL_NO = UUID.fromString("00002a25-0000-1000-8000-00805f9b34fb");
+    public static final UUID UUID_CHARACTERISTIC_DEV_INFO_HW_REVISION = UUID.fromString("00002a27-0000-1000-8000-00805f9b34fb");
+    public static final UUID UUID_CHARACTERISTIC_DEV_INFO_SW_REVISION = UUID.fromString("00002a28-0000-1000-8000-00805f9b34fb");
+    public static final UUID UUID_CHARACTERISTIC_DEV_INFO_SYSTEM_ID = UUID.fromString("00002a23-0000-1000-8000-00805f9b34fb");
+    public static final UUID UUID_CHARACTERISTIC_DEV_INFO_PNP_ID = UUID.fromString("00002a50-0000-1000-8000-00805f9b34fb");
```

Comment:
These are standard BLE UUIDs of the Device Info Profile, and we already handle at least some of them. The PNP ID for example, is not properly handled yet. It's not really obvious how we do that, so no wonder you didn't see that.

We have a class AbstractBleProfile with several subclasses that provide a reusable implementations of several standard Bluetooth profiles/services. The DeviceSupport classes then just create the profile instances and delegate the gatt events to them.

So, if you want to improve or further implement the standard profiles, please go ahead :-)

Can you rework this PR, because the button handling is good :-)